internal/controller/credential_controller.go

```diff
@@ -1,277 +0,0 @@
-// Copyright 2024
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-package controller
-
-import (
-	"context"
-	"errors"
-	"fmt"
-	"time"
-
-	corev1 "k8s.io/api/core/v1"
-	apierrors "k8s.io/apimachinery/pkg/api/errors"
-	apimeta "k8s.io/apimachinery/pkg/api/meta"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
-	capz "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
-	capv "sigs.k8s.io/cluster-api-provider-vsphere/apis/v1beta1"
-	ctrl "sigs.k8s.io/controller-runtime"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-
-	kcm "github.com/K0rdent/kcm/api/v1alpha1"
-	"github.com/K0rdent/kcm/internal/utils"
-)
-
-// CredentialReconciler reconciles a Credential object
-type CredentialReconciler struct {
-	client.Client
-	SystemNamespace string
-	syncPeriod      time.Duration
-}
-
-func (r *CredentialReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Result, err error) {
-	l := ctrl.LoggerFrom(ctx)
-	l.Info("Credential reconcile start")
-
-	cred := &kcm.Credential{}
-	if err := r.Client.Get(ctx, req.NamespacedName, cred); err != nil {
-		return ctrl.Result{}, client.IgnoreNotFound(err)
-	}
-
-	if err := utils.AddKCMComponentLabel(ctx, r.Client, cred); err != nil {
-		l.Error(err, "adding component label")
-		return ctrl.Result{}, err
-	}
```

Comment:
this code MIGHT be removed if you want to mark the kcm `credentials` explicitly with labels but i'd leave it as it is right now